flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStAggregatingState.java

```diff
@@ -0,0 +1,157 @@
+import org.apache.flink.util.Preconditions;
+
+import org.rocksdb.ColumnFamilyHandle;
+
+import java.io.IOException;
+import java.util.function.Supplier;
+
+/**
+ * The implementation of {@link AggregatingState} for ForStDB.
+ *
+ * @param <K> type of key
+ * @param <IN> type of input
+ * @param <ACC> type of aggregate state
+ * @param <OUT> type of output
+ */
+public class ForStAggregatingState<K, N, IN, ACC, OUT>
+        extends AbstractAggregatingState<K, N, IN, ACC, OUT> implements ForStInnerTable<K, N, ACC> {
+
+    /** The column family which this internal value state belongs to. */
+    private final ColumnFamilyHandle columnFamilyHandle;
+
+    /** The serialized key builder which should be thread-safe. */
+    private final ThreadLocal<SerializedCompositeKeyBuilder<K>> serializedKeyBuilder;
+
+    /** The data outputStream used for value serializer, which should be thread-safe. */
+    private final ThreadLocal<DataOutputSerializer> valueSerializerView;
+
+    /** The data inputStream used for value deserializer, which should be thread-safe. */
+    private final ThreadLocal<DataInputDeserializer> valueDeserializerView;
+
+    /** The serializer for namespace. * */
+    private final ThreadLocal<TypeSerializer<N>> namespaceSerializer;
+
+    /** The default namespace if not set. * */
+    private final N defaultNamespace;
+    /* Creates a new InternalKeyedState with the given asyncExecutionController and stateDescriptor.
+     *
+     * @param stateRequestHandler The async request handler for handling all requests.
+     * @param stateDescriptor     The properties of the state.
+     */
+    public ForStAggregatingState(
+            AggregatingStateDescriptor<IN, ACC, OUT> stateDescriptor,
+            StateRequestHandler stateRequestHandler,
+            ColumnFamilyHandle columnFamily,
+            Supplier<SerializedCompositeKeyBuilder<K>> serializedKeyBuilderInitializer,
+            N defaultNamespace,
+            Supplier<TypeSerializer<N>> namespaceSerializerInitializer,
+            Supplier<DataOutputSerializer> valueSerializerViewInitializer,
+            Supplier<DataInputDeserializer> valueDeserializerViewInitializer) {
+        super(stateRequestHandler, stateDescriptor);
+        this.columnFamilyHandle = columnFamily;
+        this.serializedKeyBuilder = ThreadLocal.withInitial(serializedKeyBuilderInitializer);
+        this.namespaceSerializer = ThreadLocal.withInitial(namespaceSerializerInitializer);
+        this.defaultNamespace = defaultNamespace;
+        this.valueDeserializerView = ThreadLocal.withInitial(valueDeserializerViewInitializer);
+        this.valueSerializerView = ThreadLocal.withInitial(valueSerializerViewInitializer);
+    }
+
+    @Override
+    public ColumnFamilyHandle getColumnFamilyHandle() {
+        return columnFamilyHandle;
+    }
+
+    @Override
+    public byte[] serializeKey(ContextKey<K, N> key) throws IOException {
+        return key.getOrCreateSerializedKey(
```

Comment:
Sry, I didn't find this util class. Could you please double check?

Reply:
It was introduced in this commit: https://github.com/apache/flink/commit/82582b3a7b75b7ffae9d48895088189b1324caa3

Reply:
@fredia, I see, I didn't pull the latest commit. So if this PR is not merged today, I will fix that tonight. Otherwise I will create another PR, since the CI takes about 6 hours. I hope it will not block your release.

Reply:
👍 No worries, you can fix it at a time convenient for you.